internal/scriptrun/scriptrun.go

```diff
@@ -0,0 +1,197 @@
+
+		if fail := venv.Activate(); fail != nil {
+			logging.Errorf("Unable to activate state: %s", fail.Error())
+			return fail.WithDescription("error_state_run_activate")
+		}
+
+		env, err := venv.GetEnv(true, filepath.Dir(proj.Source().Path()))
+		if err != nil {
+			return err
+		}
+		subs.SetEnv(env)
+
+		// search the "clean" path first (PATHS that are set by venv)
+		env, err = venv.GetEnv(false, "")
+		if err != nil {
+			return err
+		}
+		venvExePath = env["PATH"]
+	}
+
+	lang := language.Unknown
+	if len(script.Languages()) == 0 {
+		warning := locale.Tl(
+			"run_warn_deprecated_script_without_language",
+			"Scripts without a defined language currently fall back to using the default shell for your platform. This fallback mechanic will soon stop working and a language will need to be explicitly defined for each script. Please configure the '[ACTIONABLE]language[/RESET]' field with a valid option (one of [ACTIONABLE]{{.V0}}[/RESET])",
+			strings.Join(language.RecognizedNames(), ", "),
+		)
+		out.Notice(output.Heading(locale.Tl("deprecation_warning", "Deprecation Warning!")))
+		out.Notice(warning)
+
+		lang = language.MakeByShell(subs.Shell())
+	}
+
+	var attempted []string
+	for _, l := range script.Languages() {
+		execPath := l.Executable().Name()
+		searchPath := venvExePath
+		if l.Executable().CanUseThirdParty() {
+			searchPath = searchPath + string(os.PathListSeparator) + os.Getenv("PATH")
+		}
+
+		logging.Debug("Checking for %s on %s", execPath, searchPath)
+		if pathProvidesExec(searchPath, execPath) {
+			lang = l
+			logging.Debug("Found %s", execPath)
+			break
+		}
+		attempted = append(attempted, l.String())
+	}
+
+	if script.Standalone() && !lang.Executable().CanUseThirdParty() {
+		return FailStandaloneConflict.New("error_state_run_standalone_conflict")
+	}
+
+	if lang == language.Unknown {
+		if len(attempted) > 0 {
+			return locale.NewInputError(
+				"err_run_unknown_language_fallback",
+				"The language for this script is not supported or not available on your system. Attempted script execution with: {{.V0}}. Please configure the 'language' field with an available option (one, or more, of: {{.V1}})",
+				strings.Join(attempted, ", "),
+				strings.Join(language.RecognizedNames(), ", "),
+			)
+		}
+		return locale.NewInputError(
+			"err_run_unknown_language",
+			"The language for this script is not supported or not available on your system. Please configure the 'language' field with a valid option (one, or more, of: {{.V0}})", strings.Join(language.RecognizedNames(), ", "),
+		)
+	}
+
+	scriptBlock, err := script.Value()
+	if err != nil {
+		return locale.WrapError(err, "err_run_scriptval", "Could not get script value.")
+	}
+
+	sf, fail := scriptfile.New(lang, script.Name(), scriptBlock)
+	if fail != nil {
+		return fail.WithDescription("error_state_run_setup_scriptfile")
+	}
+	defer sf.Clean()
+
+	out.Notice(output.Heading(locale.Tl("script_output", "Script Output")))
```

Comment:
So it's still going to print all this stuff even when it's just running events?

Reply:
I wasn't sure about this part - what should be printed for different cases. Depending on the level of detail needed, we will end up needing to maintain two very similar functions.  Alternately, we can provide a silent outputer, or accept the output (wouldn't the user want to know the script output?).